src/v/cloud_storage/types.cc

```diff
@@ -507,32 +507,43 @@ configuration::get_bucket_config() {
     }
 }
 
-std::ostream& operator<<(std::ostream& os, upload_object_type upload) {
+std::ostream& operator<<(std::ostream& os, upload_type upload) {
     switch (upload) {
-    case upload_object_type::object:
+        using enum cloud_storage::upload_type;
+    case object:
         return os << "object";
-    case upload_object_type::segment_index:
+    case segment_index:
         return os << "segment-index";
-    case upload_object_type::manifest:
+    case manifest:
         return os << "manifest";
-    case upload_object_type::group_offsets_snapshot:
+    case group_offsets_snapshot:
         return os << "group-offsets-snapshot";
-    case upload_object_type::download_result_file:
+    case download_result_file:
         return os << "download-result-file";
-    case upload_object_type::remote_lifecycle_marker:
+    case remote_lifecycle_marker:
         return os << "remote-lifecycle-marker";
-    case upload_object_type::inventory_configuration:
+    case inventory_configuration:
         return os << "inventory-configuration";
     }
 }
 
-void upload_object_request::on_success(remote_probe& probe) {
+std::ostream& operator<<(std::ostream& os, download_type download) {
+    switch (download) {
+        using enum cloud_storage::download_type;
```

Comment:
would not have guessed this was allowed, but cool !

Reply:
Yeah, this is a new addition since c++20. It doesn't add value outside of such operator overloads which serve the enum class, but IMO looks a little less verbose here.

https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1099r5.html https://en.cppreference.com/w/cpp/language/enum#Using-enum-declaration